src/lib/tzcode/gen-zone-abbrevs.pl

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/perl
```

Comment:
It would be nice to describe (in a commit message or in a top-level comment) requirements for the script and how to use it:
```sh
sergeyb@pony:~/sources/MRG/tarantool$ perl ./src/lib/tzcode/gen-zone-abbrevs.pl src/lib/tzcode/zone-abbrevs.txt
Could not open 'main.zi' for reading: 'No such file or directory' at ./src/lib/tzcode/gen-zone-abbrevs.pl line 158.
```

Right now script requires `main.zi` which is not in a repository.

Reply:
Agreed that there is a lot I've put aside and not documented or automated. Will document it properly. (There is no time before RC1 for proper integration into cmake build process, and there is no yet update scenario implemented - we will return to automation here later, on the next steps)

Reply:
And answering where is `main.zi`? It's one of main artifacts created  after tzcode build process, and I copy it locally for `gen-zone-abbrevs.pl` consumption.

Reply:
I've documented perl script in the script itself (this is a perlpod way)

```diff
diff --git a/src/lib/tzcode/gen-zone-abbrevs.pl b/src/lib/tzcode/gen-zone-abbrevs.pl
index 7d3d6948a..a477d4b1d 100755
--- a/src/lib/tzcode/gen-zone-abbrevs.pl
+++ b/src/lib/tzcode/gen-zone-abbrevs.pl
@@ -1,4 +1,45 @@
 #!/usr/bin/perl
+=head1 NAME
+
+gen-zone-abbrevs.pl - generate list of timezones
+
+=head1 SYNOPSIS
+
+  gen-zone-abbrevs.pl [zone-abbrevs.txt [main.zi]] > timezones.h
+
+Options:
+
+  first argument is zone-abbrevs.txt or it's equivalent
+  second argument is main.zi of it's equivalent
+
+=head1 DESCRIPTION
+
+Reads input description of known IANA timezone abbreviations (including single-
+letter military zone names, e.g. Zulu 'Z'), and list of zones from Olson/tzcode
+main.zi file, and then generates set of macro calls defining timezone with all
+attributes.
+
+  ZONE_ABBREV(  25,    0, "Z", TZ_MILITARY)
+  ZONE_ABBREV( 112,    0, "UT", TZ_UTC|TZ_RFC)
+  ZONE_ABBREV( 128,    0, "ACT", TZ_AMBIGUOUS)
+  ZONE_ABBREV( 237,  240, "MSD", 0)
+  ZONE_ABBREV( 238,  180, "MSK", 0)
+
+  ZONE_UNIQUE(1000, "Etc/GMT")
+  ZONE_UNIQUE(1001, "Etc/UTC")
+  ZONE_UNIQUE(1030, "Europe/Minsk")
+  ZONE_UNIQUE(1032, "Europe/Moscow")
+
+One may use this generated file directly (via include in C/C++) or generate
+different artifact which could be used from other language (i.e. Go, or Python).
+We guarantee that constants assigned to particular zone will be maintained in
+future. To simplify future updates we leave some holes in list.
+
+NB! In a future, for the next versions of Olson/tzdata database, we plan to
+automate updates of generated list. At the moment we leave it as manual process.
+
+=cut
+
 use strict;
 use warnings;
 
```

Reply:
```sh
perldoc gen-zone-abbrevs.pl
```

Reply:
Nice, thanks!

Nit: extra whitespace in `single- letter`.


According to synopsis `main.zi` is optional:
```
SYNOPSIS
      gen-zone-abbrevs.pl [zone-abbrevs.txt [main.zi]] > timezones.h
```
but script requires it:
```
sergeyb@pony:~/sources/MRG/tarantool/src/lib/tzcode$ perl gen-zone-abbrevs.pl zone-abbrevs.txt 
Could not open 'main.zi' for reading: 'No such file or directory' at gen-zone-abbrevs.pl line 186.
```

> (There is no time before RC1 for proper integration into cmake build process, and there is no yet update scenario implemented - we will return to automation here later, on the next steps)

Should we submit a new issue with it and put it to backlog?

Reply:
No need to extra ticket here - Olson/tzcode compilation is planned to be integrated at the next step, when we will bring here work with tzdata rules. At this next step all artifacts (including currently missing `main.zi`) will become built during build procedure. At this step we have omitted it, and left out of scope, as we needed to have available different kind of artifacts - our own list of timezones in `timezones.h`. We claim that we plan to maintain order of zones in this list for all future releases. And that's a big deal.

Reply:
Please comment regarding error reported by script when `main.zi` is not specified.